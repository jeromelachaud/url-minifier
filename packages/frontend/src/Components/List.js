import React, { Component } from 'react'
import PropTypes from 'prop-types'
import LinkItem from './LinkItem'
import './List.css'

export default class Grid extends Component {
  render() {
    const {
      urls,
    } = this.props
    const listElement = urls.map((link, i) => (
      <LinkItem
        key={i}
        link={link}>
      </LinkItem>
    ))

    return (
      <div className="List">
        <table>
          <tbody>
            <tr>
              <th>Original URL</th>
              <th>Minified URL</th>
              <th>Number of visits</th>
              <th>Delete URL</th>
            </tr>
            {listElement}
          </tbody>
        </table>
      </div>
    )
  }
}

Grid.propTypes = {
  urls: PropTypes.array.isRequired,
}
