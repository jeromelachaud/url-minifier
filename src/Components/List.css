.List {
  overflow-x: auto;
  margin: 0 auto;
  font-size: 1rem;
}

.List th {
  color: var(--third-color);
}
