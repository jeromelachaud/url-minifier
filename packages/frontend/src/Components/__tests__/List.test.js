import React from 'react'
import { shallow } from 'enzyme'
import { List } from '../List'

const urls = []

it('renders correctly', () => {
  const tree = shallow(
    <List
      urls={urls}
    />
  )
  expect(tree).toMatchSnapshot()
})
