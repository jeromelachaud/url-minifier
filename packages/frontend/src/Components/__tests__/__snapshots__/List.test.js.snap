// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="List"
>
  <table>
    <tbody>
      <tr>
        <th>
          Original URL
        </th>
        <th>
          Minified URL
        </th>
        <th>
          Number of visits
        </th>
        <th>
          Delete URL
        </th>
      </tr>
    </tbody>
  </table>
</div>
`;
